man/man7/zpool-features.7

```diff
@@ -364,9 +364,9 @@ When this feature is enabled ZFS will use block cloning for operations like
 Block cloning allows to create multiple references to a single block.
 It is much faster than copying the data (as the actual data is neither read nor
 written) and takes no additional space.
-Blocks can be cloned across datasets under some conditions (like disabled
-encryption and equal
-.Nm recordsize ) .
+Blocks can be cloned across datasets under some conditions (like equal
+.Nm recordsize ,
+the same master encryption key, etc.).
```

Comment:
That leaves the reader wondering what other conditions need to be met.
Would it not be better to list all the required conditions?

Reply:
Its a fair question, but some of the limitations are imposed by the OS, not by ZFS, and they can vary by OS type and version.

Reply:
The list of requirements is so nontrivial that I am not sure it worth describing it in all the details.

Reply:
Maybe we can describe some "obvious" cases which don't work. Like moving a dataset to a new encryption root not sharing the same master key. But I go with @amotin it has a lot of corner cases, this list of tests for obvious tests and corner tests is getting longer and to describe all the corner cases is not worth I think. Maybe it make sense to add a little bit more like @robn said: "the same master encryption key, etc. Block Cloning across encrypted datasets can be limited through various (nontrivial) reasons depending on the OS and/or ZFS internals."
This makes clear it is possible under a lot of the right conditions and everybody knows if it works it works if not it's because of some nontrivial limitations.

Reply:
I wouldn't mention the test suite here, because the is the manpage and people/admins don't know where to find the test suite or read the test suite and find every limitation.